### PR TITLE
[UXE-3775] test: e2e domains main crud

### PR DESCRIPTION
**Pull request by peterpaulo-azion**
### What is the new behavior introduced by this PR?
* linter utils
* test crud domains
* selectors added
* data-testid in fieldDropdown component

Refactor:
* now we have a map of our products
* openProduct command open any product
* deleteProduct cy.visit() removed

### Does this PR introduce breaking changes?
- [x] No
- [ ] Yes 

### Does this PR introduce UI changes? Add a video or screenshots here.

https://github.com/aziontech/azion-console-kit/assets/109550332/36aa2c27-33f3-4ac5-968a-a9a1bba031c6

### Does it have a link on Figma?

<hr />

#### Make sure your pull request fits the checklist below (when applicable):

- [x] The issue title follows the format: [ISSUE_CODE] TYPE: TITLE
- [x] Commits are tagged with the right word (feat, test, refactor, etc)
- [ ] Application responsiveness was tested to different screen sizes
- [x] Code is formatted and linted
- [x] Tags are added to the PR

#### These changes were tested on the following browsers:
- [x] Chrome
- [x] Edge
- [x] Firefox
- [ ] Safari

### CODE REVIEW

File: cypress/e2e/domains.cy.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    cy.get(selectors.edgeApplication.nameInput).type(edgeAppName)
+    cy.get(selectors.edgeApplication.addressInput).type(`${edgeAppName}.edge.app`)
+
+    // Act
+    cy.get(selectors.form.actionsSubmitButton).click()
+
+    // Assert
+    cy.verifyToast('success', 'Your edge application has been created')
+    cy.get(selectors.domains.pageTitle(edgeAppName)).should('have.text', edgeAppName)
+
```
Comment (alisson-marcelo-azion) [inline cypress/e2e/domains.cy.js:24]: Provável que todo esse trecho se torne uma função auxiliar ou comando nos próximos PRs nesse teste. Mas por ora está bom 😉

File: cypress/e2e/domains.cy.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      'Succesfully created!',
+      'The domain is now available in the Domain management section.'
+    )
+    cy.get(selectors.form.actionsCancelButton).click()
+    cy.get(selectors.list.searchInput).type(domainName)
+    cy.get(selectors.list.actionsMenu.button).click()
+    cy.get(selectors.domains.actionDelete).click()
+    cy.get(selectors.list.deleteDialog.confirmationInputField).clear()
+    cy.get(selectors.list.deleteDialog.confirmationInputField).type('delete{enter}')
+    cy.verifyToast('Resource successfully deleted')
```
Comment (alisson-marcelo-azion) [inline cypress/e2e/domains.cy.js:53]: Aqui poderia usar o cy.deleteProduct()

File: cypress/e2e/domains.cy.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    cy.get(selectors.domains.edgeApplicationField).click()
+    cy.get(selectors.domains.dropdownFilter).type(edgeAppName)
+    cy.get(selectors.domains.edgeApplicationOption).click()
+    cy.get(selectors.domains.cnamesField).type(`${domainName}.domain.app`)
+
+    // Act
+    cy.get(selectors.form.actionsSubmitButton).click()
+
+    // Assert
+    cy.get(selectors.domains.dialogTitle).should('have.text', 'Domain has been created')
```
Comment (alisson-marcelo-azion) [inline cypress/e2e/domains.cy.js:38]: cy.VerifyToast aqui ?

File: cypress/support/selectors.js [hunk 4]
```diff
@@ -252,7 +252,7 @@ const selectors = {
       columnName: (columnName) => `[data-testid="list-table-block__column__${columnName}__row"]`
     }
   },
-  edgeDns:{
+  edgeDns: {
```
Comment (alisson-marcelo-azion) [inline cypress/support/selectors.js:255]: Não entendo porque o linter não havia bloqueado isso 😬

**Review by alisson-marcelo-azion — CHANGES_REQUESTED** [review]

File: cypress/e2e/domains.cy.js (hunk 3 above)
Comment (peterpaulo-azion) [reply cypress/e2e/domains.cy.js:38]: Não, aqui eu estou verificando o título de um modal/dialog.

Comment (alisson-marcelo-azion) [reply cypress/e2e/domains.cy.js:38]: Eu não conhecia isso 😮 : 
Screenshot 2024-07-03 at 15 24 36

**Review by william-tome — APPROVED** [review]

**Review by alisson-marcelo-azion — APPROVED** [review]

**Pull request merged**